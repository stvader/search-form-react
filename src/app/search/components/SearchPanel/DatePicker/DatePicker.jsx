import React from 'react';
import { instanceOf, func } from 'prop-types';
import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';

import { DATE_FORMAT, MIN_DATE } from '../../../search.dateTimeConstants';

import s from '../search-panel.module.scss';

const DatePicker = ({ selectDate, handleDateChange }) => (
  <MuiPickersUtilsProvider utils={DateFnsUtils}>
    <KeyboardDatePicker
      disableToolbar
      format={DATE_FORMAT}
      minDate={MIN_DATE}
      className={s.input}
      variant="inline"
      inputVariant="outlined"
      id="date-picker-inline"
      label="Please select date"
      value={selectDate}
      onChange={handleDateChange}
      autoOk
      KeyboardButtonProps={{ 'aria-label': 'change date' }}
      InputAdornmentProps={{ position: 'start' }}
    />
  </MuiPickersUtilsProvider>
);

DatePicker.propTypes = {
  selectDate: instanceOf(Date),
  handleDateChange: func,
};

export default DatePicker;
