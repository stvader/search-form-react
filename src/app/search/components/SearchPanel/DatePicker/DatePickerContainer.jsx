import React, { memo, useContext } from 'react';

import { SearchPanelContext } from '../SearchPanelContext';
import DatePicker from './DatePicker';

const DatePickerContainer = () => {
  const { selectDate, handleDateChange } = useContext(SearchPanelContext);

  return <DatePicker selectDate={selectDate} handleDateChange={handleDateChange} />;
};

export default memo(DatePickerContainer);
