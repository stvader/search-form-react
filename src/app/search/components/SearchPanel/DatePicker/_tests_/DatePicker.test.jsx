import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { KeyboardDatePicker } from '@material-ui/pickers';

import DatePicker from '../DatePicker';

configure({ adapter: new Adapter() });

const mockHandleDateChange = jest.fn();

describe('DatePicker', () => {
  it('DatePicker render', () => {
    const component = shallow(
      <DatePicker selectDate="Some date" onChange={mockHandleDateChange} />,
    );

    const input = component.find(KeyboardDatePicker);

    expect(component).toMatchSnapshot();
    expect(input).toBeDefined();
    expect(input.prop('value')).toBe('Some date');
  });
});
