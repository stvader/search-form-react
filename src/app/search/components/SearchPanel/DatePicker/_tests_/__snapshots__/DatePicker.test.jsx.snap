// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker DatePicker render 1`] = `ShallowWrapper {}`;
